### PR TITLE
 Separate UCI from help text.

### LINKED ISSUE #353: Separate UCI option text from command line help text
Currently UCI option names are also used as command line help. We need to separate them.

This will allow more descriptive help and standard UCI options.
It will also allow for a particular option to be command line parameter only (by not defining UCI option name).

### PR DESCRIPTION
**Pull request by mooskagh**
* Also make UCI options case insensitive.
* --futile-move-aversion is now --smart-pruning-factor
* Also write better help to some of options (more is welcome!)
* Also StrSplit had a bug

Fixes #353.

### CODE REVIEW

**Comment by borg323** [comment]
Noticed some changes in flag behavior, don't know if it is intentional:
1. -w is now -n
2. -t and -l are not accepted
3. --debuglog is now --debug-log

**Comment by borg323** [comment]
Similarly in `selfplay` mode, `-p` is not recognized while `-w` is retained.

File: src/utils/string.cc [hunk 1]
```diff
@@ -90,4 +90,32 @@ std::string Trim(std::string str) {
...
+std::vector<std::string> FlowText(const std::string& src, int width) {
+  std::vector<std::string> result;
+  auto paragraphs = StrSplit(src, "\n");
+  for (const auto& paragraph : paragraphs) {
+    result.emplace_back();
+    auto words = StrSplit(paragraph, " ");
+    for (const auto& word : words) {
+      if (result.back().empty()) {
+        // First word in line, always add.
+      } else if (result.back().size() + word.size() + 1 > width) {
```
[the remark this answers is not in the document]
Comment (mooskagh) [reply src/utils/string.cc]: Done.

**Comment by mooskagh** [comment]
`-t` and `-w` were mistakes, fixed
`-l` (and also `-p` and `-v` in selfplay mode) were intentionally deleted.
They don't seen to be important enough to me to deserve a single letter command line flag.

Maybe we need ones for backend and backend-opts though.

**Comment by borg323** [comment]
I missed -s for syzygy, but can leave it for later if anybody complains, together with short options for backend and backend opts

**Comment by mooskagh** [comment]
I don't like the formatting either. Let's hope someone will fix it one day. :)

**Pull request merged**